macos/snd_.ksy

```diff
@@ -0,0 +1,201 @@
+    42: freq_cmd
+    43: amp_cmd
+    44: timbre_cmd
+    60: wave_table_cmd
+    61: phase_cmd
+    80: sound_cmd
+    81: buffer_cmd
+    82: rate_cmd
+  sound_header_type:
+    0x00: standard
+    0xFF: extended
+    0xFE: compressed
+seq:
+  - id: format
+    type: u2
+  - id: number_of_data_formats
+    type: u2
+    if: format==1
+  - id: data_format
+    type: data_format
+    repeat: expr
+    repeat-expr: number_of_data_formats
+    if: format==1
+  - id: reference_count
+    type: u2
+    if: format==2
+  - id: number_of_sound_commands
+    type: u2
+  - id: sound_command
+    type: sound_command
+    repeat: expr
+    repeat-expr: number_of_sound_commands
+types:
+  data_format:
+    seq:
+      - id: id
+        type: u2
+        enum: data_type
+      - id: options
+        type: u4
+  sound_command:
+    instances:
+      cmd:
+        value: raw_cmd&0x7FFF
+        enum: cmd_type
+      is_data_offset:
+        value: raw_cmd&0x8000
+      sound_header_type:
+        pos: param2 + 20
+        type: u1
+        enum: sound_header_type
+        if: is_data_offset > 0 and cmd == cmd_type::buffer_cmd
+      sound_header:
+        pos: param2
+        size-eos: true
+        if: is_data_offset > 0 and cmd == cmd_type::buffer_cmd
+        type:
+          switch-on: sound_header_type
+          cases:
+            sound_header_type::standard: standard_sound_header
+            sound_header_type::extended: extended_sound_header
+            sound_header_type::compressed: compressed_sound_header
+    seq:
+      - id: raw_cmd
+        type: u2
+      - id: param1
+        type: u2
+      - id: param2
+        type: u4
+  standard_sound_header:
+    seq:
+      - id: sample_ptr
+        type: u4
+      - id: lenght
+        type: u4
+      - id: sample_rate
+        type: u2
+      - id: sample_rate_frac
+        type: u2
+      - id: loop_start
+        type: u4
+      - id: loop_end
+        type: u4
+      - id: encode
+        type: u1
+        enum: sound_header_type
+      - id: base_frequency
+        type: u1
+      - id: sample_area
+        size: lenght
+  extended_sound_header:
+    seq:
+      - id: sample_ptr
+        type: u4
+      - id: num_channels
+        type: u4
+      - id: sample_rate
+        type: u2
+      - id: sample_rate_frac
+        type: u2
+      - id: loop_start
+        type: u4
+      - id: loop_end
+        type: u4
+      - id: encode
+        type: u1
+        enum: sound_header_type
+      - id: base_frequency
+        type: u1
+      - id: num_frames
+        type: u4
+      - id: aiff_sample_rate
+        size: 10
+        #this is a float80
+      - id: marker_chunk
+        type: u4
+      - id: instrument_chunks
+        type: u4
+      - id: aes_recording
+        type: u4
+      - id: sample_size
+        type: u2
+      - id: future_use_1
+        type: u2
+      - id: future_use_2
+        type: u4
+      - id: future_use_3
+        type: u4
+      - id: future_use_4
+        type: u4
```

Comment:
It might be simpler to declare this as a single byte array - if all of the fields are for future use only, it doesn't matter what their data types are. Also, by convention fields that are reserved for future use should be called `reserved`.

```suggestion
      - id: reserved
        size: 14
```

Reply:
Done.